### PR TITLE
Gradle should fail resolution from subsequent repositories on critical errors

### PR DESCRIPTION
**Pull request by bmuschko**
### Context

See https://github.com/gradle/gradle/issues/3398 for more information

### CODE REVIEW

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy [hunk 1]
```diff
@@ -243,9 +243,10 @@ class DependencyUnresolvedModuleIntegrationTest extends AbstractHttpDependencyRe
         !downloadedLibsDir.isDirectory()
 
         where:
-        reason                          | abort | action              | outcome
-        'exceeds timeout'               | true  | 'expectGetBlocking' | 'assertDependencyMetaDataReadTimeout'
-        'returns internal server error' | false | 'expectGetBroken'   | 'assertDependencyMetaDataInternalServerError'
+        reason                          | abort | action                | outcome
+        'exceeds timeout'               | true  | 'expectGetBlocking'   | 'assertDependencyMetaDataReadTimeout'
+        'returns internal server error' | true  | 'expectGetBroken'     | 'assertDependencyMetaDataInternalServerError'
+        'returns uncritical error'      | false | 'expectGetUnofficial' | 'assertDependencyMetaDataSiteFrozenError'
```
Comment (oehme) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy]: ❌ All 5xx are critical. We should use something like 401 as an example of "uncritical".

File: subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java [hunk 2]
```diff
@@ -155,6 +160,11 @@ private CloseableHttpResponse processResponse(String source, String method, Clos
         }
         if (!wasSuccessful(response)) {
             LOGGER.info("Failed to get resource: {}. [HTTP {}: {}]", method, response.getStatusLine(), source);
+
+            if (wasUnrecoverable5xx(response)) {
```
Comment (oehme) [inline subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java]: ❌ Why introduce a new type? We already have one for http errors and we can add something like `isSeverError`

Comment (bmuschko) [reply subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java]: done

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy (hunk 1 above)
Comment (bmuschko) [reply subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy]: done

**Pull request merged**